app/subscriber/src/components/sidebar/hooks/useFilterOptions.ts

```diff
@@ -1,64 +1,88 @@
+  talkRadio: ISourceModel[];
+  onlinePrint: ISourceModel[];
+  newsRadio: ISourceModel[];
+  television: ISourceModel[];
+  sources: ISourceModel[];
+}
+
+const defaultMediaTypeGroups: IMediaTypeGroups = {
+  dailyPrint: [],
+  weeklyPrint: [],
+  cpWire: [],
+  talkRadio: [],
+  onlinePrint: [],
+  newsRadio: [],
+  television: [],
+  sources: [],
+};
+
+export const useFilterOptions = (): IMediaTypeGroups => {
+  const [{ isReady, sources, settings }] = useLookup();
+  const [mediaTypeGroups, setMediaTypeGroups] = React.useState<IMediaTypeGroups>({
+    ...defaultMediaTypeGroups,
+    sources,
+  });
 
   React.useEffect(() => {
-    if (settings.length) {
-      if (!settings.find((s) => s.name === 'MediaIds'))
+    if (isReady) {
+      var advancedSearchMediaTypeGroups = settings.find(
+        (s) => s.name === Settings.AdvancedSearchMediaTypeGroups,
+      )?.value;
+      if (advancedSearchMediaTypeGroups) {
+        try {
+          const values = JSON.parse(advancedSearchMediaTypeGroups);
+          setMediaTypeGroups({
+            dailyPrint: sources.filter(
+              (source) => source.mediaTypeSearchGroupId === values.dailyPrintId,
+            ),
+            weeklyPrint: sources.filter(
+              (source) => source.mediaTypeSearchGroupId === values.weeklyPrintId,
+            ),
+            cpWire: sources.filter((source) => source.mediaTypeSearchGroupId === values.cpWireId),
+            talkRadio: sources.filter(
+              (source) => source.mediaTypeSearchGroupId === values.talkRadioId,
+            ),
+            onlinePrint: sources.filter(
+              (source) => source.mediaTypeSearchGroupId === values.onlinePrintId,
+            ),
+            newsRadio: sources.filter(
+              (source) => source.mediaTypeSearchGroupId === values.newsRadioId,
+            ),
+            television: sources.filter(
+              (source) => source.mediaTypeSearchGroupId === values.televisionId,
+            ),
+            sources,
+          });
+        } catch {
+          toast.error(
+            `Configuration '${Settings.AdvancedSearchMediaTypeGroups}' is invalid, please contact support to set these through the editor app.`,
+          );
+        }
+      } else
         toast.error(
-          'MediaIds not found, please contact support to set these through the editor app.',
+          `Configuration '${Settings.AdvancedSearchMediaTypeGroups}' not found, please contact support to set these through the editor app.`,
         );
-      if (!settings.find((s) => s.name === 'PrintIds'))
-        toast.error(
-          'PrintIds not found, please contact support to set these through the editor app.',
-        );
-
-      setMediaIds(JSON.parse(settings.find((s) => s.name === 'MediaIds')?.value ?? '{}'));
-      setPrintIds(JSON.parse(settings.find((s) => s.name === 'PrintIds')?.value ?? '{}'));
     }
-  }, [settings]);
+  }, [isReady, settings, sources]);
 
-  const dailyPrint = sources.filter(
```

Comment:
@jdtoombs Place all this logic in the useEffect or it will cause a re-render each time.